### ISSUE TITLE
Can't get custom configuration to run

### ISSUE THREAD

**Issue by wfhartford** [mr-ransel/addon-nebula#8]
Hi, I'm trying to get nebula working on my home assistant. I have a nebula network set up already with a CA & Lighthouse, so I don't want my HA to be either. I've customized a config.yaml file based on the one generated by the add on at startup, but the add on won't start. The logs show that it's picked up my configuration, but fail at an iptables command:

```
-----------------------------------------------------------
 Add-on: Nebula
 HomeAssistant addon for slackhq/nebula
-----------------------------------------------------------
 Add-on version: dev
 You are running the latest version of this add-on.
 System: Home Assistant OS 12.3  (aarch64 / yellow)
 Home Assistant Core: 2024.5.5
 Home Assistant Supervisor: 2024.05.1
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
s6-rc: info: service base-addon-banner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service base-addon-log-level: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service base-addon-log-level successfully started
s6-rc: info: service legacy-cont-init: starting
cont-init: info: running /etc/cont-init.d/config.sh
[17:22:23] WARNING: Nebula add-on is not configured as certificate authority. You must generate and place certificates for this node and the CA in /ssl/nebula/nodes/homeassistant/homeassistant.(crt|key) and /ca/ca.crt
Generating a config.yaml
[17:22:24] NOTICE: Custom nebula config.yaml detected, ignoring generated nebula configuration!
Setting up IP Forwarding and iptables rules...
iptables v1.8.9 (legacy): host/network `null' not found
Try `iptables -h' or 'iptables --help' for more information.
cont-init: info: /etc/cont-init.d/config.sh exited 2
cont-init: warning: some scripts exited nonzero
s6-rc: warning: unable to start service legacy-cont-init: command exited 1
/run/s6/basedir/scripts/rc.init: warning: s6-rc failed to properly bring all the services up! Check your logs (in /run/uncaught-logs/current if you have in-container logging) for more information.
/run/s6/basedir/scripts/rc.init: fatal: stopping the container.
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service base-addon-log-level: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service base-addon-log-level successfully stopped
s6-rc: info: service base-addon-banner: stopping
s6-rc: info: service base-addon-banner successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped
```

My config file is here: [config.yaml](https://github.com/mr-ransel/addon-nebula/files/15449595/config.yaml.txt)

Any help would be greatly appreciated. Thanks.

**Comment by mr-ransel** [comment]
2 Quick questions:
- Does it work with a non-customized YAML, just a generated one?
- Do you know what the names of your hardware's network interfaces are?

Without any testing, I'm guessing there's a bug because the interface names are hard coded [here](https://github.com/mr-ransel/addon-nebula/blob/c240756b80413efc792e96997be68ae6e703b6c9/nebula/rootfs/etc/cont-init.d/config.sh#L194), but I will fiddle with my own instance to confirm I can reproduce the exact error once my kid is asleep. If I'm right it should be a fairly simple fix.

**Comment by mr-ransel** [comment]
Ohhh no it's definitely going to be [this line](https://github.com/mr-ransel/addon-nebula/blob/c240756b80413efc792e96997be68ae6e703b6c9/nebula/rootfs/etc/cont-init.d/config.sh#L224) because like 218 is returning 'null'. Hmm I'll work out a better override there, but the easy fix for now is to add a single item (the first item) to your addon-config node_list with the 'overlay_ip' value of your hass instance's desired nebula IP. I'll come up with a better way to do that later, but right now it just assumes the first in the list is hass, and uses its overlay IP for the iptables rule. 

You want something like this for your node_list, even if you already have your certs generated and signed elsewhere
```
- name: homeassistant-addon
  overlay_ip: 192.168.99.1
```

**Comment by wfhartford** [comment]
Thanks, that worked. I ran into one more problem which I don't really understand, but was able to work around. With the iptables problem fixed, it started nebula, but I got an error message that nebula couldn't find the ca.crt file at `/ssl/nebula/nodes/ca/ca.crt`, which is not where the ca.crt file is, and not where the config file says it is. I put my ca.crt file at `/ssl/nebula/nodes/homeassistant/ca.crt`, and put that in the config file. I'm not sure why nebula would be looking in nodes/ca. I copied the ca.crt to the nodes/ca directory and nebula start just fine.

Thanks for your help.

**Comment by mr-ransel** [comment]
Ahhh yeah it made an assumption that bring-your-own -configs would use the same file structure. These are both addressable source of truth problems. The first I'll address by getting the the overlay IP directly from the nebula cert file, and steal the path for that from the config.yaml (generated or no), and for the second , I'll add a case for bring your own config and point that validation check at whatever path is in the config.yaml (and similarly, for any others that might not make sense).

**Comment by wfhartford** [comment]
Thanks @mr-ransel. Nebula is working on my home assistant now, but it seems to have limited connectivity. I'm kind of confused since in my limited experience, nebula works pretty smoothly once a node finds the lighthouse. I'm wondering if you have seen any issues like this:

* The two servers that share a LAN with HA can connect to HA and get it's home page.
* HA cannot connect to either of the servers on the same LAN.
* My phone cannot connect to HA wither when on the same LAN or when connected to the cell network.

Any thoughts? Do you think using the host network as suggested by @d4l3k in #7 would help?

Thanks again for your help.

**Comment by wfhartford** [comment]
Thanks for your help. I got everything working once I set up relaying correctly. My phone can't connect to anything without relaying, and I forgot to add my relay node to my lighthouse config.

**Comment by zsQ8ymRSY8B4owpdCvTRdkz8rQUK** [comment]
I wanted to add that I'm also attempting to utilize a custom config (running certs and lighthouse outside of HA). I ran into the same issues as wfhartford. Based on the issues the addon has finding certs, it almost appears as if it's loading both the generated_config and config.yaml in the base folder simultaneously or in some weird combination.

I was able to get HA to handshake with the lighthouse but cannot ping from one to the other or to any other hosts. Logs show an attempt but "handshake timed out" error.
